### PR TITLE
rpcserver: add missed and live tickets rpc tests.

### PR DESCRIPTION
**Pull request by dnldd**
This adds tests for the missed and live ticket rpcs as well as the validate address rpc.

Work towards #2069.

### CODE REVIEW

File: rpcserverhandlers_test.go [hunk 1]
```diff
@@ -3488,6 +3488,80 @@ func TestHandleGetTxOutSetInfo(t *testing.T) {
...
+	t.Parallel()
+
+	tkt1, err := chainhash.NewHashFromStr("1f6631957b4060d81ba7e760ec9c8150ba028eb051ddadf2b9749a5ccda1a955")
+	if err != nil {
+		t.Fatalf("[NewHashFromStr] unexpected error: %v", err)
+	}
+	tkt2, err := chainhash.NewHashFromStr("eca7e802590df60f7d300b6170f63dfab213b26421ed2e70de3ec2224cb9e460")
+	if err != nil {
+		t.Fatalf("[NewHashFromStr] unexpected error: %v", err)
+	}
```
Comment (rstaudt2) [inline rpcserverhandlers_test.go:3501]: You can the `mustParseHash` helper function that is defined in this file (and in the next test below as well)

File: rpcserverhandlers_test.go [hunk 2]
```diff
@@ -3488,6 +3488,80 @@ func TestHandleGetTxOutSetInfo(t *testing.T) {
...
+
+	testRPCServerHandler(t, []rpcTest{{
+		name:    "handleLiveTickets: no live tickets",
+		handler: handleLiveTickets,
+		cmd:     &types.LiveTicketsCmd{},
+		mockChain: func() *testRPCChain {
+			return defaultMockRPCChain()
+		}(),
```
Comment (rstaudt2) [inline rpcserverhandlers_test.go:3509]: No need to specify `mockChain` if it's just using the default (in `TestHandleMissedTickets` as well)

File: rpcserverhandlers_test.go [hunk 3]
```diff
@@ -3488,6 +3488,80 @@ func TestHandleGetTxOutSetInfo(t *testing.T) {
...
+		cmd:     &types.LiveTicketsCmd{},
+		mockChain: func() *testRPCChain {
+			chain := defaultMockRPCChain()
+			chain.liveTickets = []chainhash.Hash{*tkt1, *tkt2}
+			return chain
+		}(),
+		result: types.LiveTicketsResult{
+			Tickets: []string{tkt1.String(), tkt2.String()},
+		},
+	}})
```
Comment (rstaudt2) [inline rpcserverhandlers_test.go:3525]: Missing a test to hit the case where `s.cfg.Chain.LiveTickets()` returns an err (can see other test cases that mock an err for an example). And similarly for `s.cfg.Chain.MissedTickets()` in the tests for `TestHandleMissedTickets`

File: rpcserverhandlers_test.go [hunk 4]
```diff
@@ -3731,6 +3811,43 @@ func TestHandleSubmitBlock(t *testing.T) {
...
+		name:    "handleValidateAddress: ok",
+		handler: handleValidateAddress,
+		cmd: &types.ValidateAddressCmd{
+			Address: "DcdacYDf5SUH5dYyDxufRngdiaVhi6n83ka",
+		},
+		result: types.ValidateAddressChainResult{
+			IsValid: true,
+			Address: "DcdacYDf5SUH5dYyDxufRngdiaVhi6n83ka",
+		},
+	},
```
Comment (davecgh) [inline rpcserverhandlers_test.go:3827]: Looks like the formatting is off here.  Need to bring the brace up.

File: rpcserverhandlers_test.go [hunk 5]
```diff
@@ -3731,6 +3811,43 @@ func TestHandleSubmitBlock(t *testing.T) {
...
+		{
+			name:    "handleValidateAddress: invalid address",
+			handler: handleValidateAddress,
+			cmd: &types.ValidateAddressCmd{
+				Address: "invalid",
+			},
+			result: types.ValidateAddressChainResult{
+				IsValid: false,
+			},
+		},
```
Comment (davecgh) [inline rpcserverhandlers_test.go:3837]: Same.

File: rpcserverhandlers_test.go [hunk 6]
```diff
@@ -3731,6 +3811,43 @@ func TestHandleSubmitBlock(t *testing.T) {
...
+		{
+			name:    "handleValidateAddress: wrong network",
+			handler: handleValidateAddress,
+			cmd: &types.ValidateAddressCmd{
+				Address: "Ssn23a3rJaCUxjqXiVSNwU6FxV45sLkiFpz",
+			},
+			result: types.ValidateAddressChainResult{
+				IsValid: false,
+			},
+		},
```
Comment (davecgh) [inline rpcserverhandlers_test.go:3847]: Same.

**Pull request merged**